.github/workflows/prerelease.yml

```diff
@@ -0,0 +1,166 @@
+name: Prerelease
+on:
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        description: 'Tag name for release'
+        required: false
+        default: prerelease
+  #push:
+  #  branches: [ main, feature/* ]
+
+
+jobs:
+
+  generate_artifact_toolkit:
+    strategy:
+      matrix:
+        supported_versions: [ '2023.3', '2023.2', '2023.1' ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'corretto'
+          java-version: '17'
+      - uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '6.x'
+      - name: Generate artifact
+        run: |
+          ./gradlew -PideProfileName=${{ matrix.supported_versions }} :plugin-toolkit:intellij:buildPlugin --exclude-task buildResharperPlugin
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: plugin-toolkit-${{ matrix.supported_versions }}
+          path: ./plugins/toolkit/intellij/build/distributions/*.zip
+          retention-days: 1
+
+  generate_artifact_q:
```

Comment:
as a future optimization, we can combine these all into a single job and just run `./gradlew buildPlugin`
as-is we're repeatedly downloading tens of gigs of dependencies

we will need to disable that task on the subprojects where it does not make sense

Reply:
Yeah, that sounds good, wanted to keep this separate currently cause:
1) Their creation is independent of each other 
2) the running out of disk space issue